File: leoz-node/src/main/java/org/deku/leoz/node/messaging/auth/v1/AuthorizationMessage.java
package org.deku.leoz.node.messaging.auth.v1;

import java.io.Serializable;

/**
 * Authorization message retrieved by node clients.
 * Created by masc on 30.06.15.
 */
public class AuthorizationMessage implements Serializable {
    private static final long serialVersionUID = 941655435886909768L;

    Integer mId;
    String mKey;
    Boolean mAuthorized;

    public Integer getId() {
        return mId;
    }

    public void setId(Integer id) {
        mId = id;
    }

    public String getKey() {
        return mKey;
    }

    public void setKey(String key) {
        mKey = key;
    }

    public Boolean getAuthorized() {
        return mAuthorized;
    }

    public void setAuthorized(Boolean authorized) {
        mAuthorized = authorized;
    }

    @Override
    public String toString() {
        return String.format("Authorization id [%s] key [%s] authorized [%s]",
                mId,
                mKey,
                mAuthorized);
    }
}

File: leoz-node/src/main/java/org/deku/leoz/node/messaging/entities/AuthorizationMessage.kt
package org.deku.leoz.node.messaging.entities

import sx.io.serialization.Serializable

/**
 * Authorization message, sent to and consumed by nodes
 * Created by masc on 30.06.15.
 */
@Serializable(0xde6de342d7a635)
data class AuthorizationMessage(
        var key: String = "",
        var authorized: Boolean = false,
        /** If the key was rejected for any reason, eg. the short representation of key was a duplicate */
        var rejected: Boolean = false)
